app/api/v1/comparison/route.ts

```diff
@@ -0,0 +1,52 @@
+import type {
+  ComparisonChartRequest,
+  ComparisonChartResponse,
+} from "@/utils/types";
+
+import {
+  comparisonChartRequestSchema,
+  comparisonChartResponseSchema,
+} from "@/validation/schema";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function POST(req: NextRequest) {
+  const body: ComparisonChartRequest = await req.json();
+  const bodyValidation = comparisonChartRequestSchema.safeParse(body);
+
+  if (!bodyValidation.success) {
+    throw new Error("Failed to validate the request.");
+  }
+
+  const { id, currency, days } = body;
+
+  const fetchHead = "https://api.coingecko.com/api/v3/coins/";
+  const fetchBody = `${id}/market_chart?vs_currency=${currency}&days=${days}`;
```

Comment:
const fetchHead = "https://api.coingecko.com/api/v3/coins";
  const fetchBody = `/${id}/market_chart?vs_currency=${currency}&days=${days}`;

we usually put the / in the next string